src/control/server/init/setup_spdk.sh

```diff
@@ -5,11 +5,65 @@
 ## to be run by an unprivileged user).
 ## These sudo commands can be granted using visudo by a system administrator.
 
-set -e
+set +e
+
+thisscriptname=`basename $0`
+thisscriptpath=$(dirname $(readlink -f $0))
+
+echo "Running script: $thisscriptname/$thisscriptpath"
 
 rootdir="$(readlink -f "$(dirname "$0")")"/../..
 scriptpath="$rootdir/spdk/scripts/setup.sh"
 
+#
+# Sets appropriate permissions on /dev/vfio/* files
+#
+set_vfio_permissions()
+{
+	# make sure regular users can read /dev/vfio
+	echo "RUN: chmod /dev/vfio"
+	sudo chmod a+x /dev/vfio
+	if [ $? -ne 0 ] ; then
+		echo "FAIL"
+		return
+	fi
+	echo "OK"
+
+	# make sure regular user can access everything inside /dev/vfio
+	echo "RUN: chmod /dev/vfio/*"
+	sudo chmod 0666 /dev/vfio/*
+	if [ $? -ne 0 ] ; then
+		echo "FAIL"
+		return
+	fi
+	echo "OK"
+
+	# since permissions are only to be set when running as
+	# regular user, we only check ulimit here
+	#
+	# warn if regular user is only allowed
+	# to memlock <64M of memory
+	MEMLOCK_AMNT=`ulimit -l`
+
+	if [ "$MEMLOCK_AMNT" != "unlimited" ] ; then
+		MEMLOCK_MB=`expr $MEMLOCK_AMNT / 1024`
```

Comment:
(lint) Use $(..) instead of legacy `..`. [SC2006]
(lint) expr is antiquated. Consider rewriting this using $((..)), ${} or [[ ]]. [SC2003]
(lint) Double quote to prevent globbing and word splitting. [SC2086]